File: src/java/ironfist/ui/jcurses/Create.java
package ironfist.ui.jcurses;

import ironfist.CreateCommand;
import ironfist.Creature;
import jcurses.event.ValueChangedEvent;
import jcurses.event.ValueChangedListener;
import jcurses.system.CharColor;
import jcurses.system.Toolkit;
import jcurses.widgets.GridLayoutManager;
import jcurses.widgets.Label;
import jcurses.widgets.Window;


/**
 * DOCUMENT ME!
 * 
 * @author pmgremo
 */
public class Create implements ValueChangedListener {
  private InputField nameField;
  private Window nameWindow;

  /**
   * DOCUMENT ME!
   */
  public void run() {
    CharColor colors = new CharColor(CharColor.BLACK, CharColor.WHITE);
    nameWindow = new Window(Toolkit.getScreenWidth(), Toolkit.getScreenHeight(), 
                            false, null);

    GridLayoutManager layout = new GridLayoutManager(2, 1);
    nameWindow.getRootPanel().setPanelColors(colors);
    nameWindow.getRootPanel().setLayoutManager(layout);
    nameWindow.setShadow(false);

    Label label = new Label("Enter Name:");
    label.setColors(colors);
    layout.addWidget(label, 0, 0, 1, 1, GridLayoutManager.ALIGNMENT_CENTER, 
                     GridLayoutManager.ALIGNMENT_RIGHT);

    nameField = new InputField(20);
    nameField.setTextComponentColors(colors);
    nameField.setDelimiter(' ');
    nameField.setDelimiterColors(colors);
    nameField.addListener(this);
    layout.addWidget(nameField, 1, 0, 1, 1, GridLayoutManager.ALIGNMENT_CENTER, 
                     GridLayoutManager.ALIGNMENT_LEFT);

    nameWindow.show();
  }

  /**
   * @see jcurses.event.ValueChangedListener#valueChanged(ValueChangedEvent)
   */
  public void valueChanged(ValueChangedEvent event) {
    Object source = event.getSource();

    if (nameField.equals(source) && nameField.isDone()) {
      String name = ((InputField) nameField).getText();
      nameWindow.close();

      Creature hero = (Creature) (new CreateCommand().execute(name));
      Game game = new Game(hero);
      game.run();
    }
  }
}

File: src/main/java/ukalus/ui/jcurses/Create.kt
package ukalus.ui.jcurses

import ukalus.CreateCommand
import ukalus.Creature
import jcurses.event.ValueChangedEvent
import jcurses.event.ValueChangedListener
import jcurses.system.CharColor
import jcurses.system.Toolkit
import jcurses.widgets.GridLayoutManager
import jcurses.widgets.Label
import jcurses.widgets.WidgetsConstants
import jcurses.widgets.Window

/**
 * DOCUMENT ME!

 * @author pmgremo
 */
class Create : ValueChangedListener {

    private var nameField: InputField? = null
    private var nameWindow: Window? = null

    /**
     * DOCUMENT ME!
     */
    fun run() {
        val colors = CharColor(CharColor.BLACK, CharColor.WHITE)
        nameWindow = Window(Toolkit.getScreenWidth(),
                Toolkit.getScreenHeight(), false, null)

        val layout = GridLayoutManager(2, 1)
        nameWindow!!.rootPanel.panelColors = colors
        nameWindow!!.rootPanel.layoutManager = layout
        nameWindow!!.setShadow(false)

        val label = Label("Enter Name:")
        label.colors = colors
        layout.addWidget(label, 0, 0, 1, 1, WidgetsConstants.ALIGNMENT_CENTER,
                WidgetsConstants.ALIGNMENT_RIGHT)

        nameField = InputField(20)
        nameField!!.setTextComponentColors(colors)
        nameField!!.setDelimiter(' ')
        nameField!!.delimiterColors = colors
        nameField!!.addListener(this)
        layout.addWidget(nameField, 1, 0, 1, 1, WidgetsConstants.ALIGNMENT_CENTER,
                WidgetsConstants.ALIGNMENT_LEFT)

        nameWindow!!.show()
    }

    /**
     * @see jcurses.event.ValueChangedListener.valueChanged
     */
    override fun valueChanged(event: ValueChangedEvent) {
        val source = event.source

        if (nameField == source && nameField!!.isDone) {
            val name = nameField!!.text
            nameWindow!!.close()

            val hero = CreateCommand().execute(name) as Creature
            val game = Game(hero)
            game.run()
        }
    }
}